src/thermostats/Thermostat.cpp

```diff
@@ -0,0 +1,92 @@
+#include <array>
+#include "Thermostat.h"
+#include "utils/ArrayUtils.h"
+
+
+
+Thermostat::Thermostat(double targetT, int periode, double maxDelta, int dimension) : target_T(targetT), periode(periode),
+                                                                                    maxDelta(maxDelta),
+                                                                                    dimension(dimension) {}
+
+void Thermostat::apply(std::vector<Particle> &particles) const {
+
+    // compute the current temperature
+
+    double T = temperature(particles, dimension);
+
+    // If the current temperature is 0, the thermostat can't be applied.
+
+    if (std::abs(T) < 1e-12) {
+        return;
+    }
+
+    // compute the temperature after the application
+
+    double TNew = T + 2 * ((target_T > T) - 0.5) * std::min(std::abs(target_T - T), maxDelta);
```

Comment:
The `2 * ((target_T > T) - 0.5)` is a bit ugly. I would instead use the [Ternary operator](https://www.geeksforgeeks.org/cpp-ternary-or-conditional-operator/):
```suggestion
    double TNew = T + (target_T > T ? 1 : - 1) * std::min(std::abs(target_T - T), maxDelta);
```
or better still
```suggestion
    double TNew = target_T > T ? T + std::min(target_T - T, maxDelta) : T - std::min(T- target_T, maxDelta);
```